idseq_dag/steps/generate_alignment_viz.py

```diff
@@ -415,13 +358,13 @@ def get_sequence_by_accession_id_s3(accession_id, entry, nt_bucket, nt_key):
                 # Such issues should be investigated. However, the pipeline step can still complete with the missing data.
                 log.write("ntDbIndexError: Failed to get nt sequence by accession ID"
                           f"{accession_id} {range_start} {range_end} {nt_bucket} {nt_key}: {e}")
-                raise e
-            except Exception as e:
+                raise
+            except:
                 if attempt + 1 < num_retries:  # Exponential backoff
-                    time.sleep(1.0 * (4**attempt))
+                    time.sleep(1.0 * (4 ** attempt))
                 else:
-                    msg = f"All retries failed for getting sequence by accession ID {accession_id}: {e}"
-                    raise RuntimeError(msg)
```

Comment:
Rather than raise a new exception, it's better practice to re-raise the current one, as it contains more detailed stacktrace, which is helpful in debugging.

If necessary, one may decorate that exception with additional data, like so:
```
except Exception as e:
    e.additional_info = "...."
    raise e
```

instead of 

```
except Exception as e:
     raise RuntimeError(f"Additional info for {e}")
```

It's one of the nice things about Python.   The first way preserves the full backtrace.

Reply:
I believe in modern Python, exception chaining is the recommended way to deal with this: `raise MyException(...) from previous_exception`.

Reply:
That's really good to know -- thanks @kislyuk